arch/risc-v/src/esp32c3/esp32c3_irq.c

```diff
@@ -371,38 +371,32 @@ IRAM_ATTR uintptr_t *esp32c3_dispatch_irq(uintptr_t mcause, uintptr_t *regs)
 
   irqinfo("INFO: mcause=%08" PRIXPTR "\n", mcause);
 
-  if ((RISCV_IRQ_BIT & mcause) != 0)
-    {
-      uint8_t cpuint = mcause & RISCV_IRQ_MASK;
-
-      DEBUGASSERT(cpuint <= ESP32C3_CPUINT_MAX);
+  DEBUGASSERT(cpuint <= ESP32C3_CPUINT_MAX);
 
-      irqinfo("INFO: cpuint=%" PRIu8 "\n", cpuint);
+  irqinfo("INFO: cpuint=%" PRIu8 "\n", cpuint);
 
+  if ((RISCV_IRQ_BIT & mcause) != 0)
```

Comment:
@no1wudi `(RISCV_IRQ_BIT & mcause) != 0` test repeats several times, how about creating a local boolean variable?

Reply:
Good idea, let's update the patch